### PR TITLE
[Keyboard][Fixes][Refactor] Capsunlocked CU80 Squish and Refactor

### LINKED ISSUE #12165: CU80 RGB MATRIX - Brightness Limits
Sorry, I missed this PR and saw it only after it was merged, and now I see some potential problems with it, which you may want to fix with a separate PR.

The brightness limiting that was done with `#define RGBLIGHT_LIMIT_VAL 50` is not handled by the RGB Matrix code — the proper option for RGB Matrix would be `RGB_MATRIX_MAXIMUM_BRIGHTNESS`. This may be important to avoid power issues.

(Note that `rgb_matrix_set_color()` bypasses the limit set by `RGB_MATRIX_MAXIMUM_BRIGHTNESS`, but that would probably be OK for the usage here, because only 2 indicator LEDs would be at the full brightness.)

LED flags are not 100% correct — 1 (`LED_FLAG_MODIFIER`) should be set for LEDs under the modifier keys like Ctrl, Shift, Alt (some effects use that flag to color the modifier keys differently from the normal ones). Setting this flag for Caps Lock and Scroll Lock looks wrong, and technically it is not even needed, because `rgb_matrix_indicators_kb()` will override the color of those LEDs irrespective of the flags used for them (except that you don't override their colors when the corresponding indicators are off, so in that case the current RGB Matrix effect would show there).

.....

BTW, speaking of keymaps, having `RGB_SPI` and `RGB_SPD` keycodes somewhere would be useful for RGB Matrix (in a pinch you can get by with just `RGB_SPI`, because holding Shift will reverse the direction of parameter change for RGB keycodes).

_Originally posted by @sigprof in https://github.com/qmk/qmk_firmware/issues/12019#issuecomment-792740732_

### PR DESCRIPTION
**Pull request by maz0r**
## Description

- Removed cu80/v2_ansi_rgb & cu80/v2_iso_rgb sub folders.

*All the below changes apply to both /v2_ansi and /v2_iso with relevant changes (num leds/ layout etc)*

- Refactor info.json to new format containing most rules.mk and config.h variables
	- Removed variables which now exist in info.json per `qmk info`

- Homologated keymaps into /default and /via using widows layout (sorry @rys  :p)

- Updated Readme.mk to include me as co-maintainer

- Refactor \<board>.c added #ifdef block to toggle RGB functionality.

- Added /rgb folder - as mentioned by @sigprof 
	- Added config.h
		- toggles CU80_RGB define used for \<board>.c code.
		- replaces product ID for VIA
		- replaces product Name for VIA
		- add undef's for underglow
		- add RGB matrix specific defines
	- Added rules.mk
		- Sets RGBLIGHT_ENABLE = no       to avoid build errors from having both lighting features enabled.
		- Sets RGB_MATRIX_ENABLE = yes    to build with rgb matrix
		- Defines the RGB_MATRIX_DRIVER = WS2812

>> This results in [WARNINGS] when /rgb firmware is build as the overrides are applied :(

image

>> VIA make all:via should also work

image

- Includes Fixes for Issue #12165 / PR #12166 rgb matrix max brightness.

*To Do*
- [x] Update Build Instructions
- [x] More testing!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

Fixes #12165 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by maz0r** [comment]
OK so it's 3am, and blowing 70mph with rain that sounds like jackhammers, so no sleep yet : (

ANSI layouts are all working, 
  - Product ID/ Name are changing as expected, so no issues with VIA picking up the board
  - Under glow animations correctly enabled and controllable with VIA / shortcuts
  - RGB Matrix animations and key press animations are working
  - Brightness limit is now working on both lighting options
  - VIA remapping is fine
  - [] Need to test sleep/wake.
  - [] Need to test with custom macro's

I don't have an ISO board to test, but the firmware loads and the offset keys/lighting is as expected.
   - Same as above but needs some basic matrix / input key map tests as I can only approximate the board with my ANSI version

**Comment by maz0r** [comment]
my bad, closed the wrong thing :/

**Comment by maz0r** [comment]
ISO testing has been carried out by @rys who has access to one of the ISO pcbs

File: keyboards/capsunlocked/cu80/v2_ansi/rgb/config.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* enable RGB matrix */
```
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_ansi/rgb/config.h:18]:
```suggestion
#pragma once

/* enable RGB matrix */
```

File: keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.c [hunk 2]
```diff
@@ -16,3 +16,56 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include "v2_ansi.h"
+
+
+#ifdef CU80_RGB
+    #include "rgb_matrix_types.h"
```
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.c:22]: could you run a format pass on this file? (`qmk cformat (filename)` should work)

File: keyboards/capsunlocked/cu80/v2_iso/keymaps/via/rules.mk [hunk 3]
```diff
@@ -1 +1,2 @@
 VIA_ENABLE = yes
+VIAL_ENABLE = yes
```
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_iso/keymaps/via/rules.mk:2]: VIAL is not supported in the repo, please remove. 
```suggestion
```

File: keyboards/capsunlocked/cu80/v2_iso/v2_iso.c [hunk 4]
```diff
@@ -16,3 +16,51 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include "v2_iso.h"
+#ifdef CU80_RGB
```
Comment (drashna) [inline keyboards/capsunlocked/cu80/v2_iso/v2_iso.c:19]: Please run a format pass on this file.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.c (hunk 2 above)
Comment (maz0r) [reply keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.c]: committed as is, after running cformat

**Review by drashna — APPROVED** [review]

File: keyboards/capsunlocked/cu80/v2_ansi/info.json [hunk 5]
```diff
@@ -1,100 +1,143 @@
...
+    "device_ver": "0x0001",
+    "pid": "0x0081",
+    "vid": "0x4355"
+  },
+  "community_layouts": ["tkl_ansi"],
+  "layouts": {
+    "LAYOUT_tkl_ansi": {
+      "c_macro": true,
+      "filename": "keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.h",
+      "key_count": 87,
```
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_ansi/info.json:51]:
```suggestion
```

File: keyboards/capsunlocked/cu80/v2_iso/info.json [hunk 6]
```diff
@@ -1,101 +1,144 @@
...
+    "device_ver": "0x0001",
+    "pid": "0x0080",
+    "vid": "0x4355"
+  },
+  "community_layouts": ["tkl_iso"],
+  "layouts": {
+    "LAYOUT_tkl_iso": {
+      "c_macro": true,
+      "filename": "keyboards/capsunlocked/cu80/v2_iso/v2_iso.h",
+      "key_count": 88,
```
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_iso/info.json:51]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.c [hunk 7]
```diff
@@ -16,3 +16,34 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include "v2_ansi.h"
+
+#ifdef CU80_RGB
+#    include "rgb_matrix_types.h"
```
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_ansi/v2_ansi.c:21]:
```suggestion
```

File: keyboards/capsunlocked/cu80/v2_iso/v2_iso.c [hunk 8]
```diff
@@ -16,3 +16,28 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+#    include "rgb_matrix_types.h"
```
Comment (fauxpark) [inline keyboards/capsunlocked/cu80/v2_iso/v2_iso.c:20]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/capsunlocked/cu80/v2_ansi/rgb/rules.mk [hunk 9]
```diff
@@ -0,0 +1,4 @@
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes     # Enable RGB matrix
+RGB_MATRIX_DRIVER = WS2812  # Set Driver to WS2812
+
```
Comment (Erovia) [inline keyboards/capsunlocked/cu80/v2_ansi/rgb/rules.mk:4]:
```suggestion
```

**Review by Erovia — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**